app/concepts/reservations/use_cases/create.rb

```diff
@@ -0,0 +1,78 @@
+module Reservations
+  module UseCases
+    class Create
+      ReservationInvalidError = Class.new(StandardError)
+
+      attr_reader :repository, :params, :ticket_desk_id
+
+      def initialize(params:, ticket_desk_id:, repository: Reservations::Repository.new)
+        @repository = repository
+        @params = params
+        @ticket_desk_id = ticket_desk_id
+      end
+
+      def call
+        Reservation.transaction do
+          repository.create(reservation_params).tap do |reservation|
+            raise ReservationInvalidError, "Couldn't create reservation" if reservation.invalid?
```

Comment:
There is problem here... `invalid?` doesn't care whenever record was actually saved to database or not - it only cared whenever object is valid in terms of validations that are described on the model. You would probably need `persisted?` or `new_record?` here - or maybe even better - to update  repository to use `create!` over `create` - that one will raise an error if it was not saved for any reason.